Azure/sonic-utilities
# Support building sonic-utilities as a Python wheel package instead of a Debian package

**- What I did**

Support building sonic-utilities as a Python wheel package rather than a Debian package.

Notes:
- Wheel packaging restricts the installation of files to within the dist-packages directory. Thus, we cannot install data files to system directories. Therefore, I am building a separate 'sonic-utilities-data' package for installing the data files (bash_completion, templates)
- Wheel packages install scripts/entrypoints to the /usr/local/bin directory, whereas Debian packages install to /usr/bin

**- How I did it**

- Specify all Python wheel dependencies in setup.py
- Remove `data_files` section from setup.py
- Build an extra 'sonic-utilities-data' Debian package which will handle installing data files to system directories
- Update all references to sonic-utilities scripts/entrypoints to either reference the new /usr/local/bin/ location or remove absolute path entirely
- Update .gitignore file as necessary

**- How to verify it**

Ensure all CLI-related functionality continues to work as expected

NOTE: PR tests will fail until the build job is updated to build as a Python wheel. PR here: https://github.com/Azure/sonic-build-tools/pull/160
